emcc.py

```diff
@@ -2008,12 +2014,8 @@ def compile_source_file(i, input_file):
 
     if link_to_object:
       with ToolchainProfiler.profile_block('linking to object file'):
-        # We have a specified target (-o <target>), which is not JavaScript or HTML, and
-        # we have multiple files: Link them
-        if shared.Settings.SIDE_MODULE:
-          exit_with_error('SIDE_MODULE must only be used when compiling to an executable shared library, and not when emitting an object file.  That is, you should be emitting a .wasm file (for wasm) or a .js file (for asm.js). Note that when compiling to a typical native suffix for a shared library (.so, .dylib, .dll; which many build systems do) then Emscripten emits an object file, which you should then compile to .wasm or .js with SIDE_MODULE.')
```

Comment:
maybe worth keeping an assert for not reaching here with SIDE_MODULE?

Reply:
I actually think is probably OK to read here with SIDE_MODULE set now.

For example I think both of these are not completely unreasonable:

`emcc -r out.o in.o -s SIDE_MODULE`
`emcc -shared libout.so in.o -s SIDE_MODULE`

Reply:
The `-r` and the `-shared` basically mean full linking is not done, but that doesn't need to mean SIDE_MODULE can't be specified.  So this change relaxed this a little.